sycl/include/sycl/handler.hpp

```diff
@@ -552,8 +558,12 @@ class __SYCL_EXPORT handler {
     // values of arguments for the kernel.
     assert(MKernel && "MKernel is not initialized");
     const std::string LambdaName = detail::KernelInfo<LambdaNameT>::getName();
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+    detail::string KernelName = getKernelName();
```

Comment:
I wonder if we can just do
```
namespace detail {
  using string = std::string;
}
```

in non-preview mode...

Reply:
I am not sure if I understand this correctly. So, correct me if I misunderstood you.
Do you meant `using detail::string = std::string;`?
Unfortunately, that is the change of ABI and it is not allowed unless you put it under #ifdef __INTEL_PREVIEW_BREAKING_CHANGES.

Reply:
Can you clarify what part of it specifically would break the ABI? The alias itself shouldn't break anything since it stands for the exact same type that's used now.

Reply:
Currently the ABI signature of getKernelName() return std::string.
I introduced detail::string for the preview mode execution path.
If we change it to return detail::string, it is an ABI break in the non-preview mode.

Reply:
I think the idea was to have the "detail/string.hpp" header provide `detail::string` as a class definition in the preview mode and as an alias for `std::string` in the non-preview one. That could significantly reduce the number of other "#ifdef __INTEL_PREVIEW_BREAKING_CHANGES" blocks that are needed since both paths would be able to just use `detail::string` to refer to their own type.

Reply:
> That could significantly reduce the number of other `#ifdef __INTEL_PREVIEW_BREAKING_CHANGES` blocks

In fact, in most cases we can't remove that conditions even if we disguise `detail::string` with `std::string`.
The reason is that `detail::string` is not simply a replacement of `std::string`.
Think about why we needed `detail::string` even if we already have `std::string`.
In the ABI-neutral cases, we **must** do conversion between detail::string_view and `std::string`, vice versa. However, this conversion is not necessary in the non-ABI-neutal cases.
So, simply disguising detail::string will not work, and still can't avoid marshalling/unmarshalling.

Reply:
@sergey-semenov 
I removed all #ifdef __INTEL_PREVIEW_BREAKING_CHANGES blocks that are not unavoidable.